File: barbecue.py
class Barbecue:
    def __init__(self, mail_server):
        self._mail_server = mail_server
        self._food = 0

    def get_food(self):
        return self._food

    def add(self, person):
        if 2 < person.get_age() <= 14:
            self._food += 100
        elif 14 < person.get_age() <= 25:
            self._food += 400
        elif person.get_age() > 25:
            self._food += 300
        
        self._mail_server.send_mail(person.get_email(), "Your participation is confirmed")

File: Barbecue.java
class Barbecue {
    private MailServer mailServer;
    private int food = 0;

    public Barbecue (MailServer mailServer) {
        this.mailServer = mailServer;
    }

    public int getFood() {
        return food;
    }

    public void add(Person p) throws InvalidEmailException {
        if (p.getAge() > 2 && p.getAge() <= 14) {
            food += 100;
        }
        else if (p.getAge() > 14 && p.getAge() <= 25) {
            food += 400;
        }
        else if (p.getAge() > 25) {
            food += 300;
        }
        this.mailServer.sendMail(p.getEmail(), "Your participation is confirmed");
    }
}
